modules/seqexec/server/src/main/scala/seqexec/server/tcs/TcsConfigRetriever.scala

```diff
@@ -193,26 +203,26 @@ object TcsConfigRetriever {
     )
   )
 
-  def retrieveConfiguration: SeqAction[EpicsTcsConfig] =
+  def retrieveConfiguration(getAoFollow: IO[Option[Boolean]]): SeqAction[EpicsTcsConfig] =
     for {
-      iaa  <- getIAA
-      offX <- getOffsetX
-      offY <- getOffsetY
-      wl   <- getWavelength
-      p1   <- getPwfs1
-      p2   <- getPwfs2
-      oi   <- getOiwfs
-      tgc  <- getGuideConfig
-      aof  <- getAoFold
-      sf   <- getScienceFoldPosition
-      hr   <- getHrwfsPickupPosition
-      ports <- getInstrumentPorts
+      iaa    <- getIAA
+      offX   <- getOffsetX
+      offY   <- getOffsetY
+      wl     <- getWavelength
+      p1     <- getPwfs1
+      p2OrAo <- getPwfs2OrAowfs(getAoFollow)
```

Comment:
This structure s used to retrieve the current TCS configuration from the actual TCS EPICS system. It has to deal with mutually exclusive P2 or AO configuration.